pkg/csi/snapshot_inspector.go

```diff
@@ -0,0 +1,252 @@
+	}
+
+	return nil
+}
+
+//go:generate go run github.com/golang/mock/mockgen -destination=mocks/mock_snapshot_browser_stepper.go -package=mocks . SnapshotBrowserStepper
+type SnapshotBrowserStepper interface {
+	ValidateArgs(ctx context.Context, args *types.SnapshotBrowseArgs) (*sv1.StorageClass, error)
+	FetchVS(ctx context.Context, args *types.SnapshotBrowseArgs) (*snapv1.VolumeSnapshot, error)
+	CreateInspectorApplication(ctx context.Context, args *types.SnapshotBrowseArgs, snapshot *snapv1.VolumeSnapshot, storageClass *sv1.StorageClass) (*v1.Pod, *v1.PersistentVolumeClaim, error)
+	PortForwardAPod(ctx context.Context, pod *v1.Pod, localPort int) error
+	Cleanup(ctx context.Context, pvc *v1.PersistentVolumeClaim, pod *v1.Pod)
+}
+
+type snapshotBrowserSteps struct {
+	validateOps          ArgumentValidator
+	versionFetchOps      ApiVersionFetcher
+	snapshotFetchOps     SnapshotFetcher
+	createAppOps         ApplicationCreator
+	portForwardOps       PortForwarder
+	cleanerOps           Cleaner
+	SnapshotGroupVersion *metav1.GroupVersionForDiscovery
+}
+
+func (s *snapshotBrowserSteps) ValidateArgs(ctx context.Context, args *types.SnapshotBrowseArgs) (*sv1.StorageClass, error) {
+	if err := args.Validate(); err != nil {
+		return nil, errors.Wrap(err, "Failed to validate input arguments")
+	}
+	if err := s.validateOps.ValidateNamespace(ctx, args.Namespace); err != nil {
+		return nil, errors.Wrap(err, "Failed to validate Namespace")
+	}
+	sc, err := s.validateOps.ValidateStorageClass(ctx, args.StorageClassName)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to validate SC")
+	}
+	groupVersion, err := s.versionFetchOps.GetCSISnapshotGroupVersion()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch groupVersion")
+	}
+	s.SnapshotGroupVersion = groupVersion
+	snapshot, err := s.validateOps.ValidateVolumeSnapshot(ctx, args.SnapshotName, args.Namespace, groupVersion)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to validate VolumeSnapshot")
+	}
+	uVSC, err := s.validateOps.ValidateVolumeSnapshotClass(ctx, *snapshot.Spec.VolumeSnapshotClassName, groupVersion)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to validate VolumeSnapshotClass")
+	}
+	vscDriver := getDriverNameFromUVSC(*uVSC, groupVersion.GroupVersion)
+	if sc.Provisioner != vscDriver {
+		return nil, fmt.Errorf("StorageClass provisioner (%s) and VolumeSnapshotClass driver (%s) are different.", sc.Provisioner, vscDriver)
+	}
+	return sc, nil
+}
+
+func (s *snapshotBrowserSteps) FetchVS(ctx context.Context, args *types.SnapshotBrowseArgs) (*snapv1.VolumeSnapshot, error) {
+	snapshotter, err := s.snapshotFetchOps.NewSnapshotter()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to load snapshotter")
+	}
+	snapArgs := &types.FetchSnapshotArgs{
+		Namespace:    args.Namespace,
+		SnapshotName: args.SnapshotName,
+	}
+	return s.snapshotFetchOps.GetVolumeSnapshot(ctx, snapshotter, snapArgs)
```

Comment:
1. Get volumesnapshot
2. Get pvc from volumesnapshotspec
3. Get storageclass from PVC and then get the storageclass to ensure it exists.
4. Return snapshot and storageClass.